### PR TITLE
Refactoring/organization of release steps

### PR DESCRIPTION
**Pull request by jviau**
- Allow for per-package release notes via `RELEASENOTES.md`
- Organize common release targets - packaging and versioning
- Update ESRP signing task, drops need for netcoreapp 2.1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,99 +1,10 @@
...
-- Durable Functions class-based syntax now resolves `ITaskActivity` instances from `IServiceProvider`, if available there.
-- `DurableTaskClient` methods have been touched up to ensure `CancellationToken` is included, as well as is the last parameter.
-- Removed obsolete/unimplemented local lambda activity calls from `TaskOrchestrationContext`
-- Input is now an optional parameter on `TaskOrchestrationContext.ContinueAsNew`
-- Multi-target gRPC projects to now use `Grpc.Net.Client` when appropriate (.NET6.0 and up)
+### Microsoft.DurableTask.Client
 
-*Note: `Microsoft.DurableTask.Generators` is remaining as `preview.1`.*
-
-## v1.0.0-rc.1
```
Comment (cgillum) [inline CHANGELOG.md:21]: I assume you're removing the pre-release notes since we're now post-GA?

File: azure-pipelines-release.yml [hunk 2]
```diff
@@ -7,18 +7,15 @@ pool:
...
 
+variables:
+  project: 'src/dirs.proj'
+
 steps:
 - checkout: self
   submodules: true
 
 - task: UseDotNet@2
-  displayName: 'Use the .NET Core 2.1 SDK (required for build signing)'
```
Comment (cgillum) [inline azure-pipelines-release.yml:15]: Was the ESPR step improved to not require .NET Core 2.1 anymore?

**Review by cgillum — COMMENTED** [review]
Just a couple questions...

File: azure-pipelines-release.yml (hunk 2 above)
Comment (jviau) [reply azure-pipelines-release.yml:15]: Yup! The `@2` version now uses 3.1, 5, or 6.

File: CHANGELOG.md (hunk 1 above)
Comment (jviau) [reply CHANGELOG.md:21]: Yeah. I think we can rely on a practice of moving release notes into GitHub released with each release to act as the larger by-release view. I am not sure if we even need to keep `CHANELOG.md` as this PR has us update `RELEASENOTES.md` on a per-project basis.

File: eng/targets/Release.targets [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <PackageReleaseNotes>$([System.IO.File]::ReadAllText('$(ReleaseNotesFile)'))</PackageReleaseNotes>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="AddCommonReleaseNotes">
+    <PropertyGroup>
+      <PackageReleaseNotes>
+See $(RepositoryUrl)/releases/tag/v$(Version)
+
+$(PackageReleaseNotes)
```
Comment (cgillum) [inline eng/targets/Release.targets:26]: This is pretty neat - I like!

**Review by cgillum — APPROVED** [review]

**Review by cgillum — APPROVED** [review]

File: src/Abstractions/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Abstractions for `Microsoft.DurableTask`. This package is for the core types used to define orchestrations and activities that are ran as part of Durable Task Framework.
+
+Commonly used types:
+- `TaskOrchestrator<TInput, TOutput>`
+- `TaskActivity<TInput, TOutput>`
+
+For more information, see https://github.com/microsoft/durabletask-dotnet/readme.md
```
[the remark this answers is not in the document]
Comment (jviau) [reply src/Abstractions/README.md]: Good catch. Considering README.md is shown on the root repo page, I decided to just link to that.

**Pull request merged**